content/05.dl-complexities.md

```diff
@@ -1,18 +1,20 @@
+One specific reproducibility pitfall that is often missed in the application of deep learning is the default use of non-deterministic algorithms by CUDA/CuDNN backends when using GPUs.
+That is, the CUDA/CuDNN architectures that facilitate the parallelized computing that power state-of-the-art DL often use algorithms by default that produce different outcomes from iteration to iteration.
+Therefore, achieving reproducibility in this context requires explicitly specifying the use of deterministic algorithms (which are typically available within your DL library), which is distinct from the setting of random seeds that typically achieve reprocubility by controlling pseudorandom deterministic procedures such as shuffling and initialization [@url:https://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#reproducibility].
 
-Similar to [Tip 4](#baselines), try to start with a relatively small network and increase the size and complexity as needed to prevent wasting time and resources. 
-Beware of the seemingly trivial choices that are being made implicitly by default settings in your framework of choice e.g. choice of optimization algorithm (adaptive methods often lead to faster convergence during training but may lead to worse generalization performance on independent datasets [@url:https://papers.nips.cc/paper/7003-the-marginal-value-of-adaptive-gradient-methods-in-machine-learning]).
-These need to be carefully considered and their impacts evaluated (see [Tip 6](#hyperparameters)).
+Similar to the suggestions in [Tip 2](#baselines) about starting with simpler models, try to start with a relatively small network, and to then increase the size and complexity as needed.
+This can help to prevent practitioners from wasting signficant time and resources on running highly complex models that feature numerous unresolved problems. 
+Again, beware of the choices that are being made implicitly (_i.e._ by default settings) by your framework of choice (e.g. choice of optimization algorithm), as these seemingly trivial specifics can actually have signficant effects on model performance.
+For example, adaptive methods often lead to faster convergence during training, but may lead to worse generalization performance on independent datasets [@url:https://papers.nips.cc/paper/7003-the-marginal-value-of-adaptive-gradient-methods-in-machine-learning]).
+These nuanced elements are easy to overlook, but it is critical to carefully consider them and to evaluate their potential impact (see [Tip 6](#hyperparameters)).
 
-In short, use smaller and simpler networks to enable faster prototyping and follow general software development best practices to maximize reproducibility.
+In short, use smaller and simpler networks to enable faster prototyping, and follow general software development best practices to maximize reproducibility.
```

Comment:
I think we do need the comma here since both verbs refer to the same subject:

```suggestion
In short, use smaller and simpler networks to enable faster prototyping and follow general software development best practices to maximize reproducibility.
```